lib/Fhaculty/Graph/Walk.php

```diff
@@ -310,4 +310,25 @@ public function isValid()
 
         return true;
     }
+
+    /**
+     * @param Walk $walk
+     */
+    public function append(Walk $walk)
+    {
+        // first edge need to be starting at the same point.
+        if(
+            $this->getVertices()->getVertexFirst() !==
+            $walk->getVertices()->getVertexLast()
+        ) {
+            throw new InvalidArgumentException('First vertex of $walk shall be the same as the last one');
+        }
+
+        $vertices = array_merge(
+            $this->getVertices()->getVector(),
+            array_slice($walk->getVertices()->getVector(), 1)
+        );
+
+        return self::factoryCycleFromVertices($vertices);
```

Comment:
This last lines creates a new `Walk` instance only for the given vertices, but ignores its edges. This means that if the graph contains parallel edges, the resulting Walk could end up with *different* edges than the original ones.

Instead, this should also merge the edges of both walks and directly pass the resulting set to its ctor.